### PR TITLE
feat(material-experimental/mdc-snackbar): Add demo.

### PR DESCRIPTION
**Pull request by opozo**
Add demo scaffolding  for MDC snackbar component

### CODE REVIEW

File: src/material-experimental/mdc-snackbar/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    name = "mdc-snackbar",
+    srcs = glob(
+        ["**/*.ts"],
+        exclude = ["**/*.spec.ts"],
+    ),
+    assets = [":snackbar_scss"] + glob(["**/*.html"]),
+    module_name = "@angular/material-experimental/mdc-snackbar",
+    deps = [
+        "@npm//material-components-web",
+    ] + CDK_TARGETS + MATERIAL_TARGETS,
```
Comment (crisbeto) [inline src/material-experimental/mdc-snackbar/BUILD.bazel]: We shouldn't be doing this. This adds all possible build targets as dependencies which will cause very slow rebuilds. We should list out the dependencies individually.

Comment (opozo) [reply src/material-experimental/mdc-snackbar/BUILD.bazel]: Done.

**Comment by jelbourn** [comment]
@opozo looks like the PR needs a rebase

**Comment by opozo** [comment]
> @opozo looks like the PR needs a rebase

Done.

**Pull request merged**